profiles/utils.py

```diff
@@ -108,3 +108,8 @@ def user_has_staff_perms_to_view_profile(
             for service_conn in service_conns
         ]
     )
+
+
+def requester_has_service_permission(request, permission):
+    service = getattr(request, "service", None)
+    return service and request.user.has_perm(permission, service)
```

Comment:
This helper is potentially called multiple times during handling of a single request. And it always hits the database. The result shouldn't change during request handling, so the result could be cached somehow.

Reply:
Implemented a cache in 8884fc8.